bb8/src/api.rs

```diff
@@ -1,15 +1,21 @@
+use crate::inner::PoolInner;
+use crate::internals::Conn;
+pub use crate::internals::State;
+use async_trait::async_trait;
 use std::borrow::Cow;
 use std::error;
 use std::fmt;
+use std::future::Future;
 use std::marker::PhantomData;
 use std::ops::{Deref, DerefMut};
+use std::pin::Pin;
 use std::time::Duration;
 
-use async_trait::async_trait;
-
-use crate::inner::PoolInner;
-use crate::internals::Conn;
-pub use crate::internals::State;
+/// An executor of futures.
+pub trait Executor: Sync + Send + 'static {
+    /// Place the future into the executor to be run.
+    fn execute(&self, fut: Pin<Box<dyn Future<Output = ()> + Send>>);
```

Comment:
Why does this need to take a `Pin<Box<_>>` instead of just an `impl Future`?

Reply:
`Executor` should be object-safe trait:
https://github.com/gaoqiangz/bb8/blob/efef42610d1a8a4d546513a621cb5af98214e7bf/bb8/src/internals.rs#L18